graphql_server/websockets/transport_ws_protocol/handlers.py

```diff
@@ -210,8 +212,12 @@ async def handle_complete(self, message: CompleteMessage) -> None:
     async def handle_invalid_message(self, error_message: str) -> None:
         await self.close(code=4400, reason=error_message)
 
-    async def send_message(self, data: TypedDict) -> None:
-        await self.send_message({**data, "type": data.type})
+    async def send_message(self, data: Message) -> None:
+        data = data.asdict()
+        assert (
+            data.get("type") is not None
+        ), "expected dict with `type` field. Got {} instead".format(data)
+        await self.send_json(data)
```

Comment:
Use dataclasses instead of TypedDicts. `TypeDict` is just a dict at runtime, and is only used for typehinting. That means `data.type` would not have worked.
sidenote: we should do the same for `graphql_ws.py` too.